### PR TITLE
first version of sequence transforms applied to Batch input

### PR DESCRIPTION
**Pull request by sararb**
Fixes # (issue)

### Goals :soccer:
- Add support for padding, transforming, and masking sequential inputs data in MM Pytorch backend
- The implemented transform classes should: 
        -  Support multiple targets 
        -  Be used for training, evaluation, and inference
 

### Implementation Details :construction:

- [ ] Implement `TabularBatchPadding` to pad a group of sequential inputs
- [ ] Implement `TabularPredictNext` for generating targets of causal next item prediction 
- [ ] Implement `TabularPredictLast` for generating targets of last item prediction 
- [ ] Implement `TabularPredictRandom` for generating targets of predicting one random item and truncate the sequence so that the random item is at the last position. 
- [ ] Implement `TabularMaskRandom` for masked language modeling training (MLM) strategy 
- [ ] Implement `TabularMaskLast` for masking last item in the sequence, generally used to evaluate models trained with MLM. 

### Testing Details :mag:
- Defined tests for padding and the different sequence transformation

### CODE REVIEW

File: merlin/models/torch/sequences.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+import torch.nn.functional as F
+from torch import nn
+
+from merlin.models.torch.batch import Batch, Sequence
+from merlin.schema import ColumnSchema, Schema, Tags
+
+MASK_PREFIX = "__mask"
+
+
+class TabularBatchPadding(nn.Module):
```
Comment (marcromeyn) [inline merlin/models/torch/sequences.py:28]: Should we rename to `TabularPadding`?

File: merlin/models/torch/sequences.py [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+from torch import nn
+
+from merlin.models.torch.batch import Batch, Sequence
+from merlin.models.torch.schema import Selection, select
+from merlin.schema import ColumnSchema, Schema, Tags
+
+MASK_PREFIX = "__mask"
+
+
+class TabularPadding(nn.Module):
```
Comment (marcromeyn) [inline merlin/models/torch/sequences.py:29]: I think it might be better to break padding out into a smaller PR first, and then do masking afterwards.
